discovery/syncer.go

```diff
@@ -833,18 +849,19 @@ func (g *GossipSyncer) genChanRangeQuery(
 	default:
 		startHeight = uint32(newestChan.BlockHeight - chanRangeQueryBuffer)
 	}
+	bestHeight := g.cfg.bestHeight()
 
 	log.Infof("GossipSyncer(%x): requesting new chans from height=%v "+
 		"and %v blocks after", g.cfg.peerPub[:], startHeight,
-		math.MaxUint32-startHeight)
+		bestHeight-startHeight)
 
 	// Finally, we'll craft the channel range query, using our starting
 	// height, then asking for all known channels to the foreseeable end of
 	// the main chain.
 	query := &lnwire.QueryChannelRange{
 		ChainHash:        g.cfg.chainHash,
 		FirstBlockHeight: startHeight,
-		NumBlocks:        math.MaxUint32 - startHeight,
+		NumBlocks:        bestHeight - startHeight,
```

Comment:
is this a change that could warrant its own commit with an explanation as well?

Reply:
Done.

Reply:
Should be fine since the graph should be synced with the chain at this point.

Reply:
it can definitely happen during a reorg

Reply:
i think it's worth checking for that case

Reply:
Done.